OpenTreeMap/otm-core
# Upgrade jQuery to 3.3.1

## Overview

Update jQuery to the latest version and jquery.fileupload to a newer version.

Connects #3228
Required by https://github.com/OpenTreeMap/otm-addons/pull/1553

### Notes

I tried to update jquery.fileupload and its companion libraries to the latest versions, but it caused a webpack error due to a change in AMD require naming.

blueimp/jQuery-File-Upload@a9bfde1

I pulled in the version before this naming change (9.12.4) and it resolved the webpack issue along with eliminating the deprecation warning from jquery-migrate.

Similarly I attempted use the latest [jquery-migrate](https://github.com/jquery/jquery-migrate/)
(3.0.1) but ran into webpack AMD errors. Rolling back to version 3.0.0 as mentioned in Github issue threads resolved the issue.

Some of our dependencies use deprecated methods. Since the methods are still available in jQuery 3.x I opted to not resolve them via updates. In some cases (typeahead) there is no compatible version to update to.

## Testing Instructions

The scope of this change is application-wide. This PR needs more of a sanity-check than a full interactive review because it will go through a full round of QA before it is released to production.

I ran through the relevant section of our QA checklist as documented in https://github.com/OpenTreeMap/otm-core/issues/3228

These are the hotspots I tested that were directly affected by upgrade changes.

* FIle uploading
  * User profile photo
  * Instance logo
  * Tree photo
  * Tree/species import
* Tree photo delete button enable status
* Management
  * Search field reordering
  * UDF choice field option editing button status
